### DISCUSSION TITLE
Adding the confirmation dialog while navigating from the core shellbar

### DISCUSSION THREAD

**Discussion by aaasarthak** [SAP/luigi#3223] [Q&A]
Scenario - 
- We have a form in the client application
- If the user tries to go on the homepage by clicking on SAP logo, switch the application or logout without saving the form data, we want to give a confirmation dialog.

Is there any approach to achieve it ?

**Comment by hardl** [comment]
Hi, have a look here: https://docs.luigi-project.io/docs/luigi-client-api/?section=setdirtystatus

**Reply by aaniket364** [reply]
Thanks for the answer, one more question though can we customize the default warning dialog box which we get if the setDirtyStatus is set to true.

**Reply by aaasarthak** [reply]
Thanks, it helped, but we are getting a console error when the user clicks on "No" in the dialog box, are we missing some configuration or is this a bug? 
![image](https://user-images.githubusercontent.com/45348322/225858514-76fd4584-010f-4708-9909-c1f8a2cd4267.png)
